homeassistant/components/mazda/binary_sensor.py

```diff
@@ -0,0 +1,184 @@
+"""Platform for Mazda binary sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
+
+
+@dataclass
+class MazdaBinarySensorRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    # Suffix to be appended to the vehicle name to obtain the binary sensor name
+    name_suffix: str
+
+    # Function to determine the value for this binary sensor, given the coordinator data
+    value: Callable[[dict], bool]
```

Comment:
Can we make `dict` more specific?

Reply:
The `dict` here is the `DataUpdateCoordinator` data. See [here](https://github.com/bdr99/core/blob/3e6b5abe649bb5a5ec7bd6737b05a05fe6f4a76b/tests/components/mazda/fixtures/diagnostics.json#L8-L60) for an example of the coordinator data for this integration. This is a complex type which includes dicts from the pymazda external library. pymazda does not export typing info, so I'm not sure how to make `dict` more specific. Can you provide any advice/suggestions? Also, the current approach matches what is used in the sensor platform for this integration.

Reply:
At the very least I think you should make this a `dict[str, Any]` as you know the keys will be strings.